add the ability to run upgrade

goneri opened the pull request.

Commit:
virtualize.sh: install /etc/ssl/certs/ci.pem

When eDeploy is used, /etc/ssl/certs/ci.pem is installed by cloud-init.
Since we now also use QCOW2, we check the file is correctlly installed.

Commit:
virtualizor: router/profile refactoring

Do not use anymore the profile of the host to know if we should add
a NATed NIC on the public_network.
Instead, we now have to configure a NIC on the machine with the folowing
keys:
 -
   nat: True
   network_name: __public_network_

It's also possible to configure the `bootproto` to get the IP from a
DHCP server. But the use of a DHCP is not mandatoried anymore.

Commit:
virtualizor.py: accept _ in prefix parameter

Commit:
virtualizor.py: increase to 2 the number of CPU

ylamgarchal committed:
Add router node

This commit adds the router node which
is the gateway of all nodes of the deployment.

Commit:
virtualize.sh: drop dup --replace parameter

virtualizor.py was called with two --replace parameters.

Commit:
virtualize.sh: ability to run upgrade

Change the argument:

    virtualize [--hypervisor=bob] version1 versio2 version2

Now it's possible to pass a list of work directory. The first one will
be deployed as usual. virtualize.sh will then run upgrade to the next
version.

During the upgrade process, the /var/lib/edeploy/conf is sourced and
RSERV and RSERV_PORT keys are used.
We had these lines manually.

Commit:
virtualize.sh: (re)write the Jenkins sudo file

In the QCOW image, the jenkins sudo file is in /etc/sudoers.d/jenkins-cloud-init.
This file is not in eDeploy exclude file and so, lost during the
upgrade.

cedlecomte commented on virtualize.sh:
```diff
@@ -21,16 +21,45 @@ set -x
 PREFIX=$USER
+installserverip=""
+usage () {
+      echo "./bob.bash [OPTION] workdir1 workdir2 etc"
```
That should be $0 not bob.bash

cedlecomte commented on virtualize.sh:
```diff
@@ -61,88 +97,118 @@ upload_logs() {
+    if [ ${do_upgrade} = 1 ]; then
+       # TODO(Gonéri): we need a better way to identify the install-server
+        ssh $SSHOPTS root@$virthost virsh destroy ${PREFIX}_os-ci-test4
+        for snapshot in $(ssh $SSHOPTS root@$virthost virsh snapshot-list --name goneri_os-ci-test4); do
```
This has to be more generic than goneri_os-ci-test4

Commit:
virtualize.sh check then destroy I-S before upgrade

Destroy the current install-server before doing the upgrade only if
the install server is really defined. This, to avoid a failure if we
run the script to time in a row.

Commit:
virtualize.sh: install-server name is a variable

Create a variable $installserver_name to store the name of the
install server. This to avoid the hardcoded `os-ci-test4` everywhere
in the code.

Commit:
virtualize.sh: update the doc

Refresh the documentation to explain the new behavior of virtualize.sh.
